src/main/java/command/CommandFactory.java

```diff
@@ -23,6 +23,23 @@ public static Command create(String userInput) throws CommandInputException {
         case HELP:
             return new HelpCommand();
 
+        case UPGRADE:
+            return new UpgradeCommand();
+
+        case HIRE:
+            String hireNum = Parser.separateCommand(userInput)[1];
+            return new HireEmployeeCommand(Integer.parseInt(hireNum));
+
+        case FIRE:
+            String fireNum = Parser.separateCommand(userInput)[1];
+            return new FireEmployeeCommand(Integer.parseInt(fireNum));
+
+        case ADJUST_SALARY:
+            String[] commandParts = Parser.separateCommand(userInput);
+            String changeType = commandParts[0].toLowerCase();
+            String amount= commandParts[1];
+            return new AdjustSalaryCommand(changeType, Integer.parseInt(amount));
+
```

Comment:
would this method be too long? are there any better ways to do the command generation?

Reply:
I think it's fine, we can add other methods to make it shorter